.github/scripts/workflow_to_docker_table.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import argparse
+import pathlib
+
+def parse_command():
+    """Function to parse the command line input
+    Args:
+        None
+    Return:
+        Namespace: returns the args as a standard Namespace object
+    """
+    parser = argparse.ArgumentParser(description='Generate Table of Tools and Their Dockers from a CWL Workflow')
+    parser.add_argument('input_file')
+    parser.add_argument('--output_file',
+        default='TABLE.md',
+        help='Path to output file.')
+    args = parser.parse_args()
+    if not args.output_file.endswith('.md'): args.output_file = args.output_file + ".md"
+    return args
+
+def parse_workflow(filepath: str, pathdict: dict | None = None) -> dict:
+    """Function to recursively build dict of unique subworkflows and tools from a file.
+    Args:
+        filepath: Path to the input cwl
+        pathdict: Dict to store the unique subworkflows and tools
+    Return:
+        pathdict: Dict containing two lists. One of unique subworkflows; one of unique tools
+    """ 
+    if not pathdict: pathdict = {}
+    if "workflows" not in pathdict: pathdict["workflows"] = []
+    if "tools" not in pathdict: pathdict["tools"] = []
+    inpath = pathlib.Path(filepath).resolve()
+    with open(inpath) as fh:
+        for line in fh:
+            if "run:" not in line: continue
+            relpath = inpath.parent / line.strip().split(' ')[-1]
+            respath = relpath.resolve()
+            if respath in pathdict["workflows"] or respath in pathdict["tools"]: continue
```

Comment:
Might have to double check, but there might be older repos in which it's just `workflow` and not `workflows`

Reply:
Yeah this doesn't really care. The only thing I truly check for is `*/tools/*` in line 40. This `pathdict` is just a custom object for this function. I could call it `pathdict["foo"]` and `pathdict["bar"]` and it wouldn't matter.